core/src/main/java/org/mule/runtime/core/api/util/StreamingUtils.java

```diff
@@ -136,4 +138,46 @@ private static void handlePossibleException(Reference<Throwable> exception) thro
   }
 
   private StreamingUtils() {}
+
+  /**
+   * Updates the value a given {@link TypedValue} instance by replacing it with a {@link CursorProvider}.
+   *
+   * @param value the typed value to update
+   * @param event the current event
+   * @param streamingManager the streaming manager
+   * @return updated {@link TypedValue instance}
+   */
+  public static TypedValue updateTypedValueForStreaming(TypedValue value, Event event, StreamingManager streamingManager) {
+    if (event == null) {
```

Comment:
This is existing code refactored, prefer not to change behaviour in something @marianogonzalez  implemented as part of a simple refactor to improve code resuse.